nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -1190,9 +1206,13 @@ in
       groups = config.users.groups;
     }) dependentCertNames;
 
-    services.nginx.additionalModules = optional cfg.recommendedBrotliSettings pkgs.nginxModules.brotli
+    services.nginx.additionalModules = lib.optional cfg.recommendedBrotliSettings pkgs.nginxModules.brotli
       ++ lib.optional cfg.recommendedZstdSettings pkgs.nginxModules.zstd;
 
+    services.nginx.hasLuaModule = lib.lists.any
+      (mod: (lib.strings.match "lua" mod.name) != null)
```

Comment:
i figured we would want this if either of `lua` or `lua-upstream` was added to `additionalModules` but i don't really know if that's correct.